### PR TITLE
Be able to use a SecretStore defined on the DDO

### PR DESCRIPTION
**Pull request by Pedro-vk**
## Description

If the DDO defiend a SecretStore, it is going to be used to be encrypted/decrypted the URLs.

## Is this PR related with an open issue?

Related to Issue https://github.com/oceanprotocol/ocean/issues/247

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 
## Checklist:

- [x] Follows the code style of this project.
- [x] Tests Cover Changes
- [x] Documentation

#### Funny gif

Put a link of a funny gif inside the parenthesis-->

### CODE REVIEW

File: src/ocean/OceanAssets.ts [hunk 1]
```diff
@@ -56,18 +56,17 @@ export default class OceanAssets {
...
-        const accessServiceDefinitionId: string = "0"
-        const computeServiceDefintionId: string = "1"
-        const metadataServiceDefinitionId: string = "2"
 
-        metadata.base.encryptedFiles = await SecretStoreProvider.getSecretStore()
-            .encryptDocument(did.getId(), metadata.base.files)
+        const authorizationService = (services.find(({type}) => type === "Authorization") || {}) as ServiceAuthorization
+        const secretStoreUrl = authorizationService.service === "SecretStore" && authorizationService.serviceEndpoint
+
+        const encryptedFiles = await SecretStoreProvider.getSecretStore(secretStoreUrl).encryptDocument(did.getId(), metadata.base.files)
```
Comment (ssallam) [inline src/ocean/OceanAssets.ts]: Need to add a default authorization service if not given in `services.

File: src/ocean/OceanAssets.ts [hunk 2]
```diff
@@ -143,11 +142,14 @@ export default class OceanAssets {
...
                             contentUrls: [],
+                            encryptedFiles,
                             files: undefined,
                         } as any,
                     },
                 },
-            ],
+                ...services
+                    .map((_) => ({..._, serviceDefinitionId: String(serviceDefinitionIdCount++)})),
+            ] as Service[],
```
Comment (ssallam) [inline src/ocean/OceanAssets.ts]: Shouldn't `services` be validated before adding them?

File: src/secretstore/SecretStoreProvider.ts [hunk 3]
```diff
@@ -8,14 +8,27 @@ export default class SecretStoreProvider {
         SecretStoreProvider.secretStore = secretStore
     }
 
-    public static getSecretStore(): SecretStore {
+    public static getSecretStore(url?: string): SecretStore {
+        if (!url) {
```
Comment (ssallam) [inline src/secretstore/SecretStoreProvider.ts]: How is an account/eth-address set on the secret store instance? when decrypting the `encryptedFiles` from the ddo, the SS must use the consumer account/password so the SS can do `.checkPermissions` on the correct consumer.

File: src/ocean/OceanAssets.ts (hunk 2 above)
Comment (Pedro-vk) [reply src/ocean/OceanAssets.ts]: No, I'll implement the validation like here: https://github.com/oceanprotocol/squid-py/blob/develop/squid_py/agreements/service_factory.py#L49-L78
Do we only support this next services? https://github.com/oceanprotocol/squid-py/blob/develop/squid_py/agreements/service_types.py

File: src/secretstore/SecretStoreProvider.ts (hunk 3 above)
Comment (Pedro-vk) [reply src/secretstore/SecretStoreProvider.ts]: @ssallam this action only works with user/password? Can it be a problem using providers like HDWallet or MetaMask?

File: src/ddo/DDO.ts [hunk 4]
```diff
@@ -46,7 +46,7 @@ export class DDO {
     public proof: Proof
 
     public constructor(ddo?: Partial<DDO>) {
-        this.created = (ddo && ddo.created) || new Date(Date.now()).toISOString().replace(/\.[0-9]{3}/, "")
+        this.created = (ddo && ddo.created) || new Date().toISOString().replace(/\.[0-9]{3}/, "")
```
Comment (ssallam) [inline src/ddo/DDO.ts]: Please check that the format matches that of the schema expected in Aquarius.

Comment (Pedro-vk) [reply src/ddo/DDO.ts]: Yes, it is ok

**Pull request merged**